llvm-project/stage1.cmake

```diff
@@ -1,45 +1,134 @@
-# TODO: comment every one of these.
+# Please try to keep these cmake variables alphabetically sorted.
+
+# Enable optimizations, as opposed to a debug build.
 set(CMAKE_BUILD_TYPE "Release" CACHE STRING "")
+
+# Use Alpine's clang and clang++ from their clang package as the stage0
+# compilers.
 set(CMAKE_CXX_COMPILER "/usr/bin/clang++" CACHE FILEPATH "")
 set(CMAKE_C_COMPILER "/usr/bin/clang" CACHE FILEPATH "")
+
+# Alpine's clang does not default to LLD; it defaults to BFD. Use LLD when
+# building executables and shared libraries.
 set(CMAKE_EXE_LINKER_FLAGS "-fuse-ld=lld" CACHE STRING "")
 set(CMAKE_SHARED_LINKER_FLAGS "-fuse-ld=lld" CACHE STRING "")
+
+# Set the default target triple to match the host.
+# TODO: passing in the value of $(clang -print-multiarch) causes failures.
+# It seems that alpine clang's default target triple is x86_64-linux-gnu.
+# Perhaps missing alpine in the triple causes some incompatibility?
 set(LLVM_DEFAULT_TARGET_TRIPLE "x86_64-alpine-linux-musl" CACHE STRING "")
+
+# Use Alpine's lld as the stage0 linker to link everything.
```

Comment:
Might want to expand why this and the `CMAKE_{EXE,SHARED}_LINKER_FLAGS` are both needed.

Reply:
Yeah, I think I didn't realize this until commenting on stage2.  I think I *might* or *should* be able to link stage1 without bfd installed. Pulling in `g++` on alpine pulls in binuilts. I think I can try removing bfd though. Then I think this should be the only cmake var necessary, and then I can drop CMAKE_EXE_LINKER_FLAGS and CMAKE_SHARED_LINKER_FLAGS for stage1. Let me try that, and see where things fall apart.

Reply:
Ok, in testing, if I hack out BFD, then `CMAKE_{EXE,SHARED}_LINKER_FLAGS` aren't needed. I'm going to retest removing all of GNU binutils, and see what breaks and whether the llvm utils will be found+used if installed and necessary.

Reply:
/bin/sh: CMAKE_AR-NOTFOUND: not found

so looks like I should:
1. install llvm13 for llvm-ar
2. remove binutils
3. set CMAKE_AR to llvm-ar

Reply:
looks like if I remove ar, I don't need to specify CMAKE_AR, cmake will find llvm-ar.